### PR TITLE
Fix log streaming

### PR DESCRIPTION
**Pull request by jerith**

### CODE REVIEW

File: seaworthy/logs.py [hunk 1]
```diff
@@ -12,6 +12,25 @@ def _last_few_log_lines(container, max_lines=100):
...
+    Return an iterator over all container logs, past and future.
+
+    The docker API we use in stream_logs() won't give us old logs, and the
+    docker API container.logs() uses seems to sometimes block indefinitely on
+    the last few lines. To get around this, we fetch all the historical logs
+    before we start streaming the new logs.
+    """
+    # Start streaming immediately after fetching the old logs to minimise the
+    # chance of a race condition.
+    old_logs = container.logs(**logs_kwargs)
```
Comment (JayH5) [inline seaworthy/logs.py]: Do we want to make sure `stream` is false?

File: seaworthy/tests-core/test_logs.py [hunk 2]
```diff
@@ -255,12 +252,76 @@ def test_streaming_waits(self):
...
+        return con
+
+    def stream(self, con, timeout=1):
+        # Always clean up the feeder machinery when we're done. This is only
+        # necessary if we timed out, but it doesn't hurt to clean up an
+        # already-clean FakeLogsContainer.
+        try:
+            return list(stream_logs(con, timeout=timeout))
+        finally:
+            con.cancel_feeder()
```
Comment (JayH5) [inline seaworthy/tests-core/test_logs.py]: Why not `con.cleanup()`?

File: seaworthy/tests-core/test_logs.py [hunk 3]
```diff
@@ -255,12 +252,76 @@ def test_streaming_waits(self):
...
+            (0.1, b'goodbye\n'),
+        ])
+        self.assertEqual(self.stream(con), [b'hello\n', b'goodbye\n'])
+        self.assertEqual(self.stream(con), [])
+        self.assertEqual(list(self.swh(con)), [b'hello\n', b'goodbye\n'])
+
+    def test_timeout_and_stream_again(self):
+        """
+        If we take too long to get the next line, we time out. When we stream
+        again, we only get historical logs as well as new ones.
```
Comment (JayH5) [inline seaworthy/tests-core/test_logs.py]: Small nit: the word "only" here doesn't seem to make sense?

Comment (jerith) [reply seaworthy/tests-core/test_logs.py]: Copy/paste fail. :-)

File: seaworthy/tests-core/test_logs.py (hunk 2 above)
Comment (jerith) [reply seaworthy/tests-core/test_logs.py]: Hysterical raisins. I've updated all the places.

File: seaworthy/logs.py (hunk 1 above)
Comment (jerith) [reply seaworthy/logs.py]: I was on the fence about that, but your question indicates that we should.

Comment (jerith) [reply seaworthy/logs.py]: Or rather, I was on the fence about rejecting the stream parameter. It needs to be `True` for `stream_logs()`, but `False` for `container.logs()`.

**Pull request merged**